server/container_create.go

```diff
@@ -351,6 +370,48 @@ func (s *Server) CreateContainer(ctx context.Context, req *pb.CreateContainerReq
 	return resp, nil
 }
 
+func (s *Server) setupOCIHooks(specgen *generate.Generator, sb *sandbox.Sandbox, containerConfig *pb.ContainerConfig, command string) error {
+	mounts := containerConfig.GetMounts()
+	for _, hook := range s.config.Hooks {
+		var (
+			match bool
+			err   error
+		)
+		logrus.Debugf("SetupOCIHooks", hook)
+		if hook.HasBindMounts && len(mounts) > 0 {
+			if err := addOCIHook(specgen, hook); err != nil {
+				return err
+			}
+			continue
+		}
+		for _, cmd := range hook.Cmds {
+			if match, err = regexp.MatchString(cmd, command); err != nil {
+				logrus.Errorf("Invalid regex %q:%q", cmd, err)
+				continue
+			}
+			if match {
+				if err := addOCIHook(specgen, hook); err != nil {
+					return err
+				}
+			}
+		}
+		for _, annotationRegex := range hook.Annotations {
```

Comment:
Is it possible that a hook might be added twice because it matches the command and annotation regexs?

Reply:
Now checking to make sure we only add once.